Tests were added

gestia535 opened the pull request.

gestia535 committed:
Tests were added

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,124 @@
+    order_history_page = OrderHistoryPage(driver)
+    return order_history_page
+
+# @pytest.fixture
+# def login(user_acc_page):
+#     login = user_acc_page.login()
+#     return login
+
+# def generate_user_credentials():
+#     email = email_generator()
+#     password = password_generator()
+#     name = create_random_name()
+#     return email, password, name
```
Нужно исправить: здесь и далее: закоммеченного кода быть не должно

AlexKitIt commented on data.py:
```diff
@@ -0,0 +1,12 @@
+BASE_URL = 'https://stellarburgers.nomoreparties.site/'
+LOGIN_URL = 'https://stellarburgers.nomoreparties.site/login'
+RESTORE_PASS_URL = 'https://stellarburgers.nomoreparties.site/forgot-password'
+RESET_PASS_URL = 'https://stellarburgers.nomoreparties.site/reset-password'
+PROFILE_URL = 'https://stellarburgers.nomoreparties.site/account/profile'
+ORDER_HISTORY_URL = 'https://stellarburgers.nomoreparties.site/account/order-history'
+REGISTER_URL = 'https://stellarburgers.nomoreparties.site/api/auth/register'
+DELETE_URL = 'https://stellarburgers.nomoreparties.site/api/auth/user'
+ORDER_URL = 'https://stellarburgers.nomoreparties.site/api/auth/orders'
```
Можно лучше: урлы стоит конкатенировать с базовым, так будет проще поддерживать несколько стендов

AlexKitIt commented on tests/test_recovery_page.py:
```diff
@@ -0,0 +1,31 @@
+        main_page.click_login_button()
+        pass_recovery_page.scroll_to_restore_button()
+        pass_recovery_page.click_restore_password_button()
+        assert driver.current_url == RESTORE_PASS_URL
```
Нужно исправить: здесь и далее: прямого обращения к driver из тестов быть не должно, только через методы пейджей

AlexKitIt commented on tests/test_recovery_page.py:
```diff
@@ -0,0 +1,31 @@
+        pass_recovery_page.set_email_field()
+        pass_recovery_page.click_reset_password_button()
+        pass_recovery_page.click_hide_password_button()
+        assert pass_recovery_page.check_visibility_passw_field_active
```
Нужно исправить: это метод но он тут не вызывается

gestia535 committed:
Исправления после ревью:
удален закомментированный код в файле conftest.py
в описании страниц password_recovery_page.py и user_account_page.py добавлены методы проверки текущего rl.
В файлы test_recovery_page.py и test_user_account_page.py исправлены вызовы методов по текущему url.

gestia535 committed:
Исправления после ревью:
удален закомментированный код в файле conftest.py
в описании страниц password_recovery_page.py и user_account_page.py добавлены методы проверки текущего rl.
В файлы test_recovery_page.py и test_user_account_page.py исправлены вызовы методов по текущему url.
Исправлен тест Проверка активации поля Пароль по клику  по кнопке показать/скрыть пароль

gestia535 committed:
Исправления после ревью:
удален закомментированный код в файле conftest.py
в описании страниц password_recovery_page.py и user_account_page.py добавлены методы проверки текущего rl.
В файлы test_recovery_page.py и test_user_account_page.py исправлены вызовы методов по текущему url.
Исправлен тест Проверка активации поля Пароль по клику  по кнопке показать/скрыть пароль

AlexKitIt commented on pages/password_recovery_page.py:
```diff
@@ -40,3 +40,11 @@ def click_hide_password_button(self):
+
+    @allure.step('Проверка, что текущий url - это url страницы восстановления пароля ')
+    def is_on_restore_password_page(self):
+        return self.driver.current_url == RESTORE_PASS_URL
```
driver.current_url описывают в base как общее действие с driver
